src/queries.test.tsx

```diff
@@ -0,0 +1,203 @@
+  addDefaultMutationFns(queryClient);
+  const TestWrapper: React.FC<TestWrapperProps> = ({ children }) => {
+    return (
+      <QueryClientProvider client={queryClient}>
+        {children}
+      </QueryClientProvider>
+    )
+  }
+  return TestWrapper
+}
+
+const TEST_ID_1 = '00000000-0000-0000-0000-000000000001'
+
+
+test('useSubmissionList should return data from the query', async () => {
+  const wrapper = createWrapper();
+  const { result } = renderHook(() => useSubmissionList({}), { wrapper })
+  await waitFor(() => expect(result.current.isSuccess).toBe(true));
+  expect(result.current.data?.results.length).toBeGreaterThan(0)
+})
+
+
+test('useSubmission should return data from the query', async () => {
+  const wrapper = createWrapper();
+  const {result} = renderHook(() => useSubmission(TEST_ID_1), { wrapper })
+  await waitFor(() => expect(result.current.query.isSuccess).toBe(true));
+  expect(result.current.query.data?.id).toBe(TEST_ID_1)
+});
+
+
+test('useSubmission should return initial data from the list query', async () => {
+  const wrapper = createWrapper();
+
+  // First fetch data from the submission list query
+  // TODO: fix hardcoded pagination options
+  const { result: listResult } = renderHook(() => useSubmissionList({ limit: 10, offset: 0 }), { wrapper })
+  await waitFor(() => expect(listResult.current.isSuccess).toBe(true));
+
+  // Second fetch data from the individual submission query. Data should be provided before waiting
+  // for the query to succeed, but a background fetch should be in progress
+  const { result: submissionResult } = renderHook(() => useSubmission(TEST_ID_1), { wrapper })
+  expect(submissionResult.current.query.isFetching).toBe(true)
+  expect(submissionResult.current.query.data?.id).toBe(TEST_ID_1)
+
+  // Data should still be provided after the background fetch completes
+  await waitFor(() => expect(listResult.current.isFetching).toBe(false));
+  expect(submissionResult.current.query.data?.id).toBe(TEST_ID_1)
+})
+
+
+test('useSubmission should mutate data', async () => {
+  const wrapper = createWrapper();
+
+  // First fetch data from the submission list query
+  // TODO: fix hardcoded pagination options
+  const { result: listResult, rerender: listRerender } = renderHook(() => useSubmissionList({ limit: 10, offset: 0 }), { wrapper })
+  await waitFor(() => expect(listResult.current.isSuccess).toBe(true));
+
+  // Second fetch data from the individual submission query. Wait for the background fetch to
+  // complete before proceeding.
+  const { result } = renderHook(() => useSubmission(TEST_ID_1), { wrapper })
+  await waitFor(() => expect(result.current.query.isFetching).toBe(false));
+
+  // Update the submission data. There doesn't seem to be a good way to ensure that the onMutate()
+  // function is called without waiting for the entire mutation to complete. To verify that the
+  // data was updated optimistically, we initiate the mutation and then introduce a very short delay
```

Comment:
Note to myself (@eecavanna): Look into this after I become more familiar with React Query.